airbyte-integrations/connectors/destination-clickhouse/src/main/java/io/airbyte/integrations/destination/clickhouse/ClickhouseDestination.java

```diff
@@ -81,26 +81,13 @@ public JsonNode toJdbcConfig(final JsonNode config) {
   }
 
   @Override
-  public AirbyteConnectionStatus check(final JsonNode config) {
-    final DataSource dataSource = getDataSource(config);
-    try {
-      final JdbcDatabase database = getDatabase(dataSource);
-      final NamingConventionTransformer namingResolver = getNamingResolver();
-      final String outputSchema = namingResolver.getIdentifier(config.get(JdbcUtils.DATABASE_KEY).asText());
-      attemptSQLCreateAndDropTableOperations(outputSchema, database, namingResolver, getSqlOperations());
-      return new AirbyteConnectionStatus().withStatus(Status.SUCCEEDED);
-    } catch (final Exception e) {
-      LOGGER.error("Exception while checking connection: ", e);
-      return new AirbyteConnectionStatus()
-          .withStatus(Status.FAILED)
-          .withMessage("Could not connect with provided configuration. \n" + e.getMessage());
-    } finally {
-      try {
-        DataSourceFactory.close(dataSource);
-      } catch (final Exception e) {
-        LOGGER.warn("Unable to close data source.", e);
-      }
+  protected AirbyteConnectionStatus checkedConnectionStatus(final DataSource dataSource, final JsonNode config, ConfiguredAirbyteCatalog catalog) throws Exception {
```

Comment:
It feels like _most_ of the modified Destinations use this same method. Can it be pulled up into a base class at all?

Reply:
I might not be following - the [clickhouse](https://github.com/airbytehq/airbyte/blob/6cffa643ca40832ab375d47eac756ed41be58bc9/airbyte-integrations/connectors/destination-clickhouse/src/main/java/io/airbyte/integrations/destination/clickhouse/ClickhouseDestination.java#L84) impl looks identical to the [abstract](https://github.com/airbytehq/airbyte/blob/6cffa643ca40832ab375d47eac756ed41be58bc9/airbyte-integrations/connectors/destination-jdbc/src/main/java/io/airbyte/integrations/destination/jdbc/AbstractJdbcDestination.java#L102) impl? (except without the try catch)